### PR TITLE
Update kernel ABI to latest FW version

### PR DESCRIPTION
**Pull request by lgirdwood**
Mostly copies and changes for license header/include guards. Will test all tomorrow.

### CODE REVIEW

**Comment by lgirdwood** [comment]
@plbossart lets reserve it - Will update both PRs

**Comment by plbossart** [comment]
even when I ignore the type issues reported by checkpatch, you still have a few (only took one per occurrence):

````text
WARNING: please, no space before tabs
#168: FILE: include/sound/sof/control.h:77:
+^ISOF_CTRL_CMD_ENUM, ^I/**< maps to ALSA enum style controls */$

WARNING: Block comments should align the * on each line
#2838: FILE: include/uapi/sound/sof/trace.h:3:
+/*
+* This file is provided under a dual BSD/GPLv2 license.  When using or

WARNING: please, no space before tabs
#168: FILE: include/sound/sof/control.h:77:
+^ISOF_CTRL_CMD_ENUM, ^I/**< maps to ALSA enum style controls */$

CHECK: Please don't use multiple blank lines
#271: FILE: include/sound/sof/dai-intel.h:49:
+
+
````
And this one is new to me.

````text
WARNING: __packed is preferred over __attribute__((packed))
#382: FILE: include/sound/sof/dai-intel.h:160:
+} __attribute__((packed));
````

**Comment by lgirdwood** [comment]
@plbossart I think these are all historical from before checkpatch. I've taken this opportunity to fix all the important ones for this update in both kernel and FW.
Please feel free to apply both FW and kernel ABI PR's if you have time tonight otherwise I can merge tomorrow if there are no new issues.

**Comment by plbossart** [comment]
@lgirdwood I'll do a smoke test with both kernel and firmware and merge both if things look fine.

**Comment by plbossart** [comment]
No issues with this PR and the matching SOF one, except for 
````text
[    5.403033] sof-audio sof-audio: ipc tx succeeded: 0x30130000: GLB_TPLG_MSG: PIPE_COMPLETE
[    5.403050] sof-audio sof-audio: error: failed to enter PM idle -11
````
Still no luck with the PM...

verified all files. The only things I noticed are
- missing __packed__ in eq.h
- whitespace differences in header.h
- missing lines in manifest.h
- trace.h (packed vs __packed__, whitespaces in comments, different definition for long_entry_address uint32_t vs. uintptr_t, spurious  ifndef at the end). As usual the trace is the worst :-(

In other words you work tomorrow... (If you merge the kernel use rebase-and-merge please, not git merge)

**Comment by ranj063** [comment]
> No issues with this PR and the matching SOF one, except for
> 
> ```
> [    5.403033] sof-audio sof-audio: ipc tx succeeded: 0x30130000: GLB_TPLG_MSG: PIPE_COMPLETE
> [    5.403050] sof-audio sof-audio: error: failed to enter PM idle -11

@plbossart do you see it even after you merged the latest PM PR #249 
> ```
> Still no luck with the PM...
> 
> verified all files. The only things I noticed are
> 
> * missing **packed** in eq.h
> * whitespace differences in header.h
> * missing lines in manifest.h
> * trace.h (packed vs **packed**, whitespaces in comments, different definition for long_entry_address uint32_t vs. uintptr_t, spurious  ifndef at the end). As usual the trace is the worst :-(
> 
> In other words you work tomorrow... (If you merge the kernel use rebase-and-merge please, not git merge)

**Comment by lgirdwood** [comment]
@plbossart I've just double checked everything and I cant see any missing packed in eq.h or problematic differences in trace.h . These were differences in an older version so I wonder if github loading was high on Friday. I've also done a git diff against my local branches and remote branches so it should all be in sync today.

**Comment by lgirdwood** [comment]
@ranj063 I had a quick check on the PM idle and we should see a dev_err() on all the suspend failures in pm.c. Error -11 is -EAGAIN and means "try again" so  its the PM core here, please look at drivers/base/power/runtime.c and rpm_suspend() as it does return -EAGAIN in two places !

**Comment by plbossart** [comment]
Try comparing your kernel and sof files, and you'll have the differences I reported, e.g.

````diff
diff eq.h /data/pbossart/SOF-public/sof-sdk/sof.git/src/include/uapi/user/eq.h 
1d0
...
62c86
< };
---
> } __attribute__((packed));
72c96
< };
---
> } __attribute__((packed));
131c155
< };
---
> } __attribute__((packed));
141c165
< };
---
> } __attribute__((packed));
151c175
< };
---
> } __attribute__((packed));
164c188
< #endif
---
> #endif /* __INCLUDE_UAPI_USER_EQ_H__ */
````

````diff
diff header.h /data/pbossart/SOF-public/sof-sdk/sof.git/src/include/uapi/user/header.h
1d0
....
25c55
< }  __attribute__((packed));
---
> } __attribute__((packed));
````

````diff
diff manifest.h /data/pbossart/SOF-public/sof-sdk/sof.git/src/include/uapi/user/manifest.h
> #include <stdint.h>
102d131
< 
````
````difff
diff trace.h /data/pbossart/SOF-public/sof-sdk/sof.git/src/include/uapi/user/trace.h 
25c50
< } __attribute__((__packed__));
---
> } __attribute__((packed));
67c92
< } __attribute__((__packed__));
---
> } __attribute__((packed));
74c99
< } __attribute__((__packed__));
---
> } __attribute__((packed));
79,83c104,108
< *  Log entry header.
< *
< * The header is followed by an array of arguments (uint32_t[]).
< * Number of arguments is specified by the params_num field of log_entry
< */
---
>  *  Log entry header.
>  *
>  * The header is followed by an array of arguments (uint32_t[]).
>  * Number of arguments is specified by the params_num field of log_entry
>  */
90,91c115,116
< 	uintptr_t log_entry_address;	/* Address of log entry in ELF */
< } __attribute__((__packed__));
---
> 	uint32_t log_entry_address;	/* Address of log entry in ELF */
> } __attribute__((packed));
93c118
< #endif
---
> #endif //#ifndef __INCLUDE_LOGGING__

````

**Comment by plbossart** [comment]
Never mind, if I fetch the latest branches again I only see differences in __packed and __attribute((packed)) which is ok. I'll merge both kernel and firmware.
